internal/driver/glfw/window_desktop.go

```diff
@@ -314,7 +314,10 @@ func (w *window) detectScale() float32 {
 		return 1.0
 	}
 
-	widthMm, _ := monitor.GetPhysicalSize()
+	widthMm, heightMm := monitor.GetPhysicalSize()
+	if widthMm == 60 && heightMm == 60 { // some sort of failure - mostly on Steam Deck
```

Comment:
Should this be not be protected by a `runtime.GOOS == "linux"` check? Maybe even `!build.IsWayland` from what I understand of the issue?

Reply:
The Wayland check is above and it will exit early.
Linux specific... I guess so, updated

Reply:
Good point :)